kolibri/plugins/user/assets/src/vue/profile-page/index.vue

```diff
@@ -1,50 +1,105 @@
 <template>
 
   <div class="profile-page">
-    <div v-if="hasPrivilege('username')" class="input-field">
-      <label>Username</label>
-      <input autocomplete="username" id="username" type="text"/>
+    <div v-if="error" class="error ">
+      {{errorMessage}}
     </div>
-    <div v-if="hasPrivilege('name')" class="input-field">
-      <label>Name</label>
-      <input autocomplete="name" id="name" type="text"/>
-    </div>
-    <div v-if="hasPrivilege('password')"  class="input-field">
-      <label>Password</label>
-      <input autocomplete="new-password" id="password" type="password"/>
-    </div>
-    <div v-if="hasPrivilege('password')"  class="input-field">
-      <label>Confirm Password</label>
-      <input autocomplete="new-password" id="confirm-password" type="password"/>
-    </div>
-    <div v-if="hasPrivilege('delete')"  class="input-field">
-      <span class="advanced-option">Delete Account</span>
-    </div>
-    <div class="input-field">
-      <button>Update Profile</button>
+    <div v-if="success" class="success">
+      {{$tr('success')}}
     </div>
+    <form @focus.capture="" @submit.prevent="submitEdits">
+      <div v-if="hasPrivilege('username')" class="input-field">
+        <label>Username</label>
+        <input v-model.lazy="username" autocomplete="username" id="username" type="text"/>
+      </div>
+      <div v-if="hasPrivilege('name')" class="input-field">
+        <label>Name</label>
+        <input v-model.lazy="full_name" autocomplete="name" id="name" type="text"/>
+      </div>
+      <div v-if="hasPrivilege('password')"  class="input-field">
+        <label>Password</label>
+        <input autocomplete="new-password" id="password" type="password"/>
+      </div>
+      <div v-if="hasPrivilege('password')"  class="input-field">
+        <label>Confirm Password</label>
+        <input autocomplete="new-password" id="confirm-password" type="password"/>
+      </div>
+      <div v-if="hasPrivilege('delete')"  class="input-field">
+        <span class="advanced-option">Delete Account</span>
+      </div>
+      <div class="input-field">
+        <button :disabled="busy" type="submit">Update Profile</button>
+      </div>
+    </form>
   </div>
 
 </template>
 
 
 <script>
 
+  const actions = require('../../actions');
+
   module.exports = {
     name: 'profile-page',
-    data: () => ({
-      username: '',
-      name: '',
-      password: '',
-    }),
+    $trNameSpace: 'profile-page',
+    $trs: {
+      genericError: 'Something went wrong',
+      success: 'Changes successfully made',
+    },
+    data() {
+      return {
+        username: this.session.username,
+        full_name: this.session.full_name,
+        password: '',
+        confirm_password: '',
+      };
+    },
+    watch: {
+      // going to be used for validation
+      // username() {return;},
+      // confirm_password() {return this.passwordsMatch();},
+    },
```

Comment:
Let's take out this block. I'm not sure that 'watch' will be the right strategy - perhaps computed properties instead? In either case, we're not implementing it yet.